src/main/java/frc/robot/CommandFactory.java

```diff
@@ -37,6 +40,65 @@ public CommandFactory(
         this.superStructure = superStructure;
     }
 
+    private boolean atTargetPosition(double currentPosition, double targetPosition, double threshold) {
+        return Math.abs(currentPosition - targetPosition) <= threshold;
+    }
+
+    /**
+     * Tells the super Structure where to go based on the Robot Mode
+     *
+     * @param robotMode A supplier for the current RobotMode so it knows where to go
+     * @return The command to follow the current position based on the Robot Mode
+     */
+    public Command goToPosition(Supplier<RobotMode> robotMode) {
+        DoubleSupplier targetRobotModeElevatorHeight = () -> switch (robotMode.get()) {
+            case CORAL_LEVEL_1 -> Constants.ArmPositions.CORAL_LEVEL_1_ELEVATOR_HEIGHT;
+            case CORAL_LEVEL_2 -> Constants.ArmPositions.CORAL_LEVEL_2_ELEVATOR_HEIGHT;
+            case CORAL_LEVEL_3 -> Constants.ArmPositions.CORAL_LEVEL_3_ELEVATOR_HEIGHT;
+            case CORAL_LEVEL_4 -> Constants.ArmPositions.CORAL_LEVEL_4_ELEVATOR_HEIGHT;
+            case ALGAE_PROCESSOR -> Constants.ArmPositions.ALGAE_PROCESSOR_ELEVATOR_HEIGHT;
+            case ALGAE_NET -> Constants.ArmPositions.ALGAE_NET_ELEVATOR_HEIGHT;
+            case ALGAE_REMOVE_LOWER -> Constants.ArmPositions.REMOVE_ALGAE_LOWER_ELEVATOR_HEIGHT;
+            case ALGAE_REMOVE_UPPER -> Constants.ArmPositions.REMOVE_ALGAE_UPPER_ELEVATOR_HEIGHT;
```

Comment:
These large switch statements make the logic harder to follow. Instead, we can add the elevator, shoulder, and wrist positions into the enum itself. See the `RobotMode` enum on how to do this.

That would allow you to do `robotMode.get().getElevatorPosition()`, etc.